Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/HttpResponse.java

```diff
@@ -56,6 +56,15 @@ public String getMessage() {
         return headers;
     }
 
+    public String findHeader(String headerName) {
+        return getHeaders().stream()
+                .filter(header -> header.getName().equalsIgnoreCase(headerName))
+                // refer to https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2
+                // combine all values by comma
+                .map(Header::getValue)
+                .collect(Collectors.joining(","));
```

Comment:
Why are we using `.collect` rather than `.findFirst().orElse(null)`. I suppose there would only be 0 or 1 header matches with the target header name. If this is by design, maybe you need to split the header by `,` at `SparkBatchSubmission::isAzureADLoginUrl`

Reply:
Just as code comments, combine all values by comma is rfc2616 design. It's a good suggestion to throw an exception for multi values.